Src/VimCore/CoreInterfaces.fs

```diff
@@ -1709,17 +1709,22 @@ type BlockSpan =
     val private _tabStop: int
     val private _spaces: int
     val private _height: int
+    val private _endOfLine: bool
 
     new(startPoint: SnapshotPoint, tabStop, spaces, height) = 
         let startColumn = VirtualSnapshotColumn(startPoint)
-        { _startColumn = startColumn; _tabStop = tabStop; _spaces = spaces; _height = height }
+        { _startColumn = startColumn; _tabStop = tabStop; _spaces = spaces; _height = height; _endOfLine = false }
 
-    new(startPoint: VirtualSnapshotPoint, tabStop, spaces, height) =
+    new(startPoint: SnapshotPoint, tabStop, spaces, height, endOfLine) = 

```

Comment:
When is it valid to have `endOfLine == true` and `spaces != 0`? Aren't these properties mutually exclusive?

Reply:
The spaces are usually non-zero when `endOfLine` is true. It represents the width in spaces of the "proxy block selection", typically the length in spaces from the anchor point to the caret point on whatever line the caret is on.

In other words, because Visual Studio won't allow us to display a ragged box selection, we have to display **some** selection, so we use a regular box selection with the width appropriate for the current line. That selection will be too narrow for some lines and too wide for others, but it will be exactly right for the caret line. So the "spaces" value (even though end-of-line is set) still represents the space width for the line the caret is on.

Reply:
Gotcha. Can we doc that somewhere? 

---
In reply to: [300842389](https://github.com/VsVim/VsVim/pull/2668#discussion_r300842389) [](ancestors = 300842389)